From 02272b731d3aa1c81ec2b9c87f0b03f7d6650cc0 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 15 Oct 2020 17:18:13 +0300
Subject: [PATCH] [SW-12754] gaudi: security: narrow includes use

For IF RTRs offsets, use only SIF, not both (SIF and NIF).
As a convention this makes it the same as other functions, since SIF
base is lower than NIF.

Change-Id: Ida1288c8238d73a5401a34304fd21319474e5616
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index ba46cc8..8f3eed7 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -25,8 +25,6 @@
 #include <sif_rtr_0_regs.h>
 #include <sif_rtr_ctrl_0_regs.h>
 #include <sif_rtr_ctrl_7_regs.h>
-#include <nif_rtr_0_regs.h>
-#include <nif_rtr_ctrl_0_regs.h>
 #include <psoc_global_conf_regs.h>
 #include <dma_if_w_s_down_ch0_regs.h>
 #include <dma_if_w_s_regs.h>
@@ -534,14 +532,14 @@
 
 	/* offsets: 0x500 - 0x580 */
 	prot_base += SNGL_REG_PROT_SIZE * 10;
-	mask = ~(PROT_BIT(mmNIF_RTR_0_LBW_RANGE_PROT_MAX_AR_8) &
-		 PROT_BIT(mmNIF_RTR_0_LBW_RANGE_PROT_MAX_AR_9) &
-		 PROT_BIT(mmNIF_RTR_0_LBW_RANGE_PROT_MAX_AR_10) &
-		 PROT_BIT(mmNIF_RTR_0_LBW_RANGE_PROT_MAX_AR_11) &
-		 PROT_BIT(mmNIF_RTR_0_LBW_RANGE_PROT_MAX_AR_12) &
-		 PROT_BIT(mmNIF_RTR_0_LBW_RANGE_PROT_MAX_AR_13) &
-		 PROT_BIT(mmNIF_RTR_0_LBW_RANGE_PROT_MAX_AR_14) &
-		 PROT_BIT(mmNIF_RTR_0_LBW_RANGE_PROT_MAX_AR_15));
+	mask = ~(PROT_BIT(mmSIF_RTR_0_LBW_RANGE_PROT_MAX_AR_8) &
+		 PROT_BIT(mmSIF_RTR_0_LBW_RANGE_PROT_MAX_AR_9) &
+		 PROT_BIT(mmSIF_RTR_0_LBW_RANGE_PROT_MAX_AR_10) &
+		 PROT_BIT(mmSIF_RTR_0_LBW_RANGE_PROT_MAX_AR_11) &
+		 PROT_BIT(mmSIF_RTR_0_LBW_RANGE_PROT_MAX_AR_12) &
+		 PROT_BIT(mmSIF_RTR_0_LBW_RANGE_PROT_MAX_AR_13) &
+		 PROT_BIT(mmSIF_RTR_0_LBW_RANGE_PROT_MAX_AR_14) &
+		 PROT_BIT(mmSIF_RTR_0_LBW_RANGE_PROT_MAX_AR_15));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	/* offsets: 0x580 - 0xF80 */
@@ -558,26 +556,26 @@
 
 	/* offsets: 0x580 - 0x600 */
 	prot_base += SNGL_REG_PROT_SIZE * 11;
-	mask = ~(PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_0) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_1) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_2) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_3) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_4) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_5) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_6));
+	mask = ~(PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_0) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_1) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_2) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_3) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_4) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_5) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AW_6));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	/* offsets: 0x600 - 0x680 allowed access */
 
 	/* offsets: 0x680 - 0x700 as following: */
 	prot_base += SNGL_REG_PROT_SIZE * 2;
-	mask = (PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_0) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_1) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_2) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_3) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_4) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_5) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_6));
+	mask = (PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_0) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_1) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_2) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_3) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_4) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_5) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_6));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	/* offsets: 0x700 - 0x800 not allowed access */
@@ -586,44 +584,44 @@
 
 	/* offsets: 0x800 - 0880 as following: */
 	prot_base += SNGL_REG_PROT_SIZE * 2;
-	mask =  ~(PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_0) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_1) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_2) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_3) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_4) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_5) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_6) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_7) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_8) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_9) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_10) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_11) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_12) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_13) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_14) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_15) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_0) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_1) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_2) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_3) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_4) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_5) &
-		 PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_6));
+	mask =  ~(PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_0) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_1) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_2) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_3) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_4) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_5) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_6) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_7) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_8) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_9) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_10) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_11) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_12) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_13) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_14) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_LOW_AR_15) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_0) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_1) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_2) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_3) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_4) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_5) &
+		 PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_BASE_HIGH_AR_6));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	/* offsets: 0x880 - 0x900 allowed access */
 
 	/* offsets: 0x900 - 0x980 as following: */
 	prot_base += SNGL_REG_PROT_SIZE * 2;
-	mask = ~(PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_7) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_8) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_9) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_10) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_11) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_12) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_13) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_14) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_15));
+	mask = ~(PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_7) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_8) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_9) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_10) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_11) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_12) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_13) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_14) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_15));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	/* offsets: 0x980 - 0xA00 not allowed access */
@@ -632,8 +630,8 @@
 
 	/* offsets: 0xA00 - 0xA80 as following: */
 	prot_base += SNGL_REG_PROT_SIZE;
-	mask = ~(PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_HIT_AW) &
-		PROT_BIT(mmNIF_RTR_CTRL_0_RANGE_SEC_HIT_AR));
+	mask = ~(PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_HIT_AW) &
+		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_HIT_AR));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	/* offsets: 0xA80 - 0xF80 */
